src/compiler/checker.ts

```diff
@@ -25844,7 +25895,22 @@ namespace ts {
          *         and 1 insertion/deletion at 3 characters)
          */
         function getSpellingSuggestionForName(name: string, symbols: Symbol[], meaning: SymbolFlags): Symbol | undefined {
-            return getSpellingSuggestion(name, symbols, getCandidateName);
+            if (!isSpellingSuggestionEdgeCase(name)) {
+                return getSpellingSuggestion(name, symbols, getCandidateName);
+            }
+
+            function isSpellingSuggestionEdgeCase(possibleEdgeCase: string) {
+                /**
+                 * Without this check, the user could be returned an unhelpful error message.
+                 * One case would be with getOwnPropertyDescriptor and getOwnPropertyDescriptors.
+                 * In this case, the user would be suggested to target getOwnPropertyDescriptor.
+                 * If he wanted to use the es2017 feature, he should be suggested to change his target lib to es2017.
+                 *
+                 */
+                const SPELLING_EDGE_CASES = ["getOwnPropertyDescriptors"];
+                return SPELLING_EDGE_CASES.includes(possibleEdgeCase);
+            }
+

```

Comment:
1. You don't need a whole function for this
2. I think you can move this check into `getSuggestedSymbolForNonexistentProperty` - not sure what @sandersn thinks

Reply:
1. Fair enough
2. I have tried implementing it in my own function, but my diagnostic would get replaced by the one generated by getSpellingSuggestionForName

Reply:
I got rid of this check entirely in my upcoming code. I feel like this is out of scope for this PR and this should be worked as a separate bug fix for `getSpellingSuggestionForName`

> Can you remove the new messages and switch the old one to accept a 2nd template slot?

I found many more edge cases when I removed the new messages and switched the old one to accept a 2nd template slot for the lib version. 

For example, all the Math functions included in es2015 such as sinh, cosh, or all other functions one letter away from an existing function would not return the proper error message if they were not all included inside of `isSpellingSuggestionEdgeCase`. 

Instead of returning `... Try changing the lib compiler to  '{1}' or later`, it would return `sinh does not exist, did you mean sin?` which might be confusing for someone who was trying to use sinh but is not using the correct configuration.

Of course, once #40320 is fixed, I am willing to work on a fix for this problem.